### ISSUE TITLE
Temperature calculation is wrong when using a DS18S20 sensor.

### ISSUE THREAD

**Issue by wecl** [arendst/Sonoff-Tasmota#5375]
see below

**Comment by netpok** [comment]
Please fill out the form above, especially the parts about your problem.

**Comment by wecl** [comment]
**To Reproduce**
connected a DS18S20 sensor to GPIO14 to a yunshan module and record a time series.

**Expected behavior**
smooth change of temperature over time.

**actual faulty behaviour**
when temperature is in- or decreasing the calculated temperature is jumping suddenly about 0.5 degrees at some point in time.

**what i did**
I looked to the sourcecode of the file xsns_05_ds18x20.ino, function Ds18x20Read() and compared the calculation with the spec of the DS18S20 sensor (https://datasheets.maximintegrated.com/en/ds/DS18S20.pdf).
On page 6 they describe as follow:
"Resolutions greater than 9 bits can be calculated using the data from the temperature, COUNT REMAIN and COUNT PER °C registers in the scratchpad. Note that the COUNT PER  °C  register  is  hard-wired  to  16  (10h).  After  reading  the  scratchpad,  the  TEMP_READ  value  is  obtained  by  truncating  the  0.5°C  bit  (bit  0)  from  the  temperature  data  (see Figure  4)..."

On line 332 of xsns_05_ds18x20.ino, the code fragment
`
if (data[0] & 1) {
          temp9 = ((data[0] >> 1) + 0.5) * sign;
        } else {
          temp9 = (data[0] >> 1) * sign;
        }
`
should be replaced by

`temp9 = (data[0] >> 1) * sign;`
because it is wrong to add 0.5 in case of calculating resolution greater than 9 bit.

I changed the code accordingly and tested it on my module and the output is as expected.
But i didn't test the calculation when the temperature is lower than 0 degrees.
